charts/leonardo/templates/cronjob.yaml

```diff
@@ -0,0 +1,91 @@
+          containers:
+          - image: "{{ .Values.cronjob.imageRepository }}:{{ .Values.cronjob.imageTag }}"
+            name: "{{ .Values.cronjob.name }}"
+            command: ["/bin/bash", "-c"]
+            args:
+              - |
+                trap "touch /tmp/pod/main-terminated" EXIT
+                /opt/docker/bin/resource-validator --dryRun --all
+            env:
+            - name: JAVA_OPTS
+              value: -Dlogback.configurationFile=file:/opt/docker/resource-validator-config/logback.xml
+            - name: LEONARDO_DB_USER
+              valueFrom:
+                secretKeyRef:
+                  name: leonardo-db-creds
+                  key: username              
+            - name: LEONARDO_DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: leonardo-db-creds
+                  key: password
+            - name: LEONARDO_PATH_TO_CREDENTIAL
+              value: /secrets/service-account.json
+            volumeMounts:
+            - name: leonardo-sa-json-file
+              mountPath: /secrets
+              readOnly: true
+            - mountPath: /tmp/pod
+              name: tmp-pod
+            - mountPath: /opt/docker/resource-validator-config
+              name: logback-config
+          - name: cloudsql-proxy
+            image: gcr.io/cloudsql-docker/gce-proxy:1.15
+            env:
+            - name: SQL_INSTANCE_PROJECT
+              valueFrom:
+                secretKeyRef:
+                  name: leonardo-db-creds
+                  key: instance_project
+            - name: SQL_INSTANCE_REGION
+              valueFrom:
+                secretKeyRef:
+                  name: leonardo-db-creds
+                  key: instance_region
+            - name: SQL_INSTANCE_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: leonardo-db-creds
+                  key: instance_name
+            command: ["/bin/sh", "-c"]
+            args:
+            - |
+              /cloud_sql_proxy --dir=/cloudsql -instances=$(SQL_INSTANCE_PROJECT):$(SQL_INSTANCE_REGION):$(SQL_INSTANCE_NAME)=tcp:0.0.0.0:3306 -credential_file=/secrets/service-account.json &
+              CHILD_PID=$!
+              (while true; do if [[ -f "/tmp/pod/main-terminated" ]]; then kill $CHILD_PID; echo "Killed $CHILD_PID as the main container terminated."; fi; sleep 1; done) &
+              wait $CHILD_PID
+              if [[ -f "/tmp/pod/main-terminated" ]]; then exit 0; echo "Job completed. Exiting..."; fi
+            volumeMounts:
+            - name: leonardo-sa-json-file
+              mountPath: /secrets
+              readOnly: true
+            - mountPath: /tmp/pod
+              name: tmp-pod
+              readOnly: true
+          volumes:
+          - name: leonardo-sa-json-file
+            secret:
+              secretName: leonardo-sa-json
+          - name: tmp-pod
+            emptyDir: {}
+          - name: logback-config
```

Comment:
not needed for cron job..but can be useful for later migrating Leonardo, hence leaving it here for now

Reply:
Actually... Will the resource-validator and the leonardo app share any config files in common? 🤔